Kernel/Devices/HID/KeyboardDevice.h

```diff
@@ -48,6 +55,9 @@ class KeyboardDevice : public HIDDevice {
     }
 
 protected:
+    RawKeyEvent handle_scan_code_input_event_set1(ScanCodeEvent);
+    RawKeyEvent handle_scan_code_input_event_set2(ScanCodeEvent);
```

Comment:
I agree with Daniel, why does this scan code set1/set2 logic need to belong on the base class? It is specific to I8042 devices, and should not be leaked into the higher level abstraction of "a keyboard with unknown physical connection to the computer"

Reply:
@supercomputer7 as far as I see it, this comment is blocking the PR. We should not be adding PS2-specific protected methods to a class that is shared b/w different keyboard devices.